include/dynamatic/Dialect/Handshake/HandshakeOps.td

```diff
@@ -395,7 +395,8 @@ def MuxOp : Handshake_Op<"mux", [
   Pure,
   VariadicHasElement<"dataOperands">,
   AllDataTypesMatchWithVariadic<"dataOperands", ["result"]>,
-  AllExtraSignalsMatchWithVariadic<"dataOperands", ["result", "selectOperand"]>,
+  AllExtraSignalsMatchWithVariadic<"dataOperands", ["result"]>, 
```

Comment:
Could you implement `AllTypesMatchWithVariadic` and replace these two constraints?

I think `AllTypesMatchWithVariadic` can be implemented in a similar (and simpler) way to `AllDataTypesMatchWithVariadic`.